### PR TITLE
Copy to localhost during solve

### PR DESCRIPTION
**Pull request by alexcb**
implements copying artifacts from containerized targets to `LOCALLY` targets. This functionality is different from `SAVE ARTIFACT AS LOCAL`, in that it is performed during the build rather than at the end of the build during the export stage.

This depends on https://github.com/earthly/buildkit/pull/30

### CODE REVIEW

File: earthfile2llb/interpreter.go [hunk 1]
```diff
@@ -268,11 +268,15 @@ func (i *Interpreter) handleRun(ctx context.Context, cmd spec.Command) error {
 			return Errorf(cmd.SourceLocation, "secrets need to be implemented for the LOCALLY directive")
 		}
 
-		err = i.converter.RunLocal(ctx, fs.Args(), *pushFlag)
-		if err != nil {
-			return WrapError(err, cmd.SourceLocation, "apply RUN")
+		if i.withDocker == nil {
```
Comment (vladaionescu) [inline earthfile2llb/interpreter.go:271]: Nit: It's easier to read if this condition is reversed - the more rare cases being nested deeper.

File: earthfile2llb/interpreter.go [hunk 2]
```diff
@@ -416,9 +420,16 @@ func (i *Interpreter) handleCopy(ctx context.Context, cmd spec.Command) error {
...
+					return WrapError(err, cmd.SourceLocation, "copy artifact locally")
+				}
+			} else {
+				err = i.converter.CopyArtifact(ctx, src, dest, platform, buildArgs.Args, *isDirCopy, *keepTs, *keepOwn, *chown, *ifExists)
+				if err != nil {
+					return WrapError(err, cmd.SourceLocation, "copy artifact")
+				}
 			}
 		}
 	} else {
```
Comment (vladaionescu) [inline earthfile2llb/interpreter.go:435]: This else case should be handled too somehow. Maybe not in this PR, but at the very least we should return an error when attempting

```
LOCALLY
COPY ./some-file ./
```

File: earthfile2llb/interpreter.go (hunk 1 above)
Comment (alexcb) [reply earthfile2llb/interpreter.go:271]: I have a follow up PR which will expand on handling the withDocker logic which is I suspect will be more complicated than the without logic. I'll swap it for now and might swap back if it helps with readability later.

File: earthfile2llb/interpreter.go (hunk 2 above)
Comment (alexcb) [reply earthfile2llb/interpreter.go:435]: that's a good point, I overlooked this; i'll change it to an error for this PR and replicate+fix later

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**